src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extension.tsx

```diff
@@ -34,6 +36,14 @@ export interface QueryEditorExtensionDependencies {
    * Set whether the query editor is collapsed.
    */
   setIsCollapsed: (isCollapsed: boolean) => void;
+  /**
+   * Currently set Query
+   */
+  query: Query;
```

Comment:
nit: what are we using this for? if it's for accessing the current query since it's with the same component it should already have access to the same query. therefore we could just access this query directly

Reply:
This is used to access the dataset on the query, passed in query to be future safe